src/Compilers/CSharp/Portable/Errors/MessageProvider.cs

```diff
@@ -118,6 +118,14 @@ public override string GetErrorDisplayString(ISymbol symbol)
             return SymbolDisplay.ToDisplayString(symbol, SymbolDisplayFormat.CSharpShortErrorMessageFormat);
         }
 
+        public override bool GetIsEnabledByDefault(int code) =>
+            (ErrorCode)code is not (
+                ErrorCode.WRN_UseDefViolationPropertyStructThis
+                or ErrorCode.WRN_UseDefViolationFieldStructThis
```

Comment:
I'm a little worried about having warnings that are disabled by default and potentially hard to discover.
From offline discussion, some possible mitigations:
- leverage an existing flag from SDK to activate those warnings
- give some unobtrusive hint in IDE that something is going on (we added a default assignment for you) and some warnings could be turned on
- confirm that some customers would really benefit from those warnings (runtime?) otherwise consider removing?

Reply:
@tannergooding @jmarolf for people who deeply care about avoiding redundant writes to structs, is there some kind of "performance sensitive" setting they can enable in their project which lights up some related analyzers, etc in the SDK? If one exists perhaps we can tie enabling these diagnostics to that setting.

Reply:
From offline discussion, it seems possible that we'd want to light these warnings up as a group if the user sets a certain value for the property [`<AnalysisLevelPerformance>`](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#analysislevelcategory).

Reply:
Looking at the values for `AnalysisMode`, perhaps this would be the necessary incantation:

```xml
<AnalysisLevelPerformance>all</AnalysisLevelPerformance>
```

Reply:
yep, we have the concept of a `Performance` analysis level. I would be fine with tying this to that in some way. Unclear to me if warning here is something that is going to be a common enough scenario that we will want to give folks a quick way to do it vs. just having them using the existing mechanisms we have to bump something up to a warning.

Reply:
Opened #60176 to track doing this. I don't think we should attempt to solve at the same time as this PR.